rust/kernel/error.rs

```diff
@@ -106,15 +106,20 @@ impl From<AllocError> for Error {
     }
 }
 
+// # Invariant: `-bindings::MAX_ERRNO` fits in an `i16`.
+crate::static_assert!(bindings::MAX_ERRNO <= -(i16::MIN as i32) as u32);
```

Comment:
I have the nagging feeling that this method is too contrived. Is there an idiomatic way to verify whether a value fits in a type? I tried `Try` or `TryFrom` but that doesn't work, because those types can't run inside a `static_assert!`.

Reply:
I have taken a look at this -- it is indeed a bit painful. After having tried several approaches, I am preparing a PR with a fun approach.

Reply:
I am grabbing the popcorn !! :)

Reply:
See if you like #269 :)

It would allow you to do:

    static_assert!((-(bindings::MAX_ERRNO as i128)) fits in i16);

Reply:
I like it a lot !! Will definitely rebase to that after it gets merged !! :100: